### ISSUE TITLE
No tray icon on KDE

### ISSUE THREAD

**Issue by arabine** [dmikushin/tray#4]
Hello,

There is no tray icon visible on KDE. I don't know where to start but I can propose a patch if someone has some hints.

**Comment by dmikushin** [comment]
Hi @arabine , thank you for reporting this. I think the issue is connected to the use of a special functional kind of icon, as explained below. I provide the following example for Gnome 3 Shell:

<img src="screenshot_linux.png">

As you can see, the icon is set to "envelope" by the following lines of code:

```
#define TRAY_ICON1 "indicator-messages"
#define TRAY_ICON2 "indicator-messages-new"
```

My guess is that these two have no valid meaning in KDE and hence not respected, and tray icon is not shown at all.

I'd start by looking into how KDE encodes its special functional icons, such as notifications. A patch should offer a more portable way to use functional icons in different desktop environments.

Alternatively, you can specify a custom bitmap icon, and check whether it works in KDE (it should work out of the box).

**Comment by arabine** [comment]
I tested with a standard icon name specified by the FreeDesktop and it works well on KDE: https://specifications.freedesktop.org/icon-naming-spec/icon-naming-spec-latest.html

![Screenshot_20230308_111945](https://user-images.githubusercontent.com/22543673/223688306-1abee89f-fc90-478c-bb47-fa3813c96bb7.png)

I started a pull-request with multiple changes, including this topic.

**Comment by dmikushin** [comment]
Hi @arabine , thank you so much for working on this! I will review the PR #8 .

**Comment by ReenigneArcher** [comment]
I discovered that you can use your own icon if you place an svg directly in `/usr/share/icons` (I didn't test PNG, maybe that works as well). Only tested on Ubuntu. I tried using a subfolder, but it didn't work... maybe just need to define the icon differently in the code.

![image](https://user-images.githubusercontent.com/42013603/224852533-0c6b80fd-a77a-4389-b44a-34ca301aaf1a.png)

Then in your CMakeLists.txt
```cmake
install(FILES "${CMAKE_SOURCE_DIR}/some-icon-name.svg" DESTINATION "/usr/share/icons")
```

Is it possible to define the full path to the image instead of just `some-icon-name`? I'd prefer to not install to that icon location if possible.


I also found this list of icons which are probably more universally supported. https://github.com/GNOME/gtk/blob/main/demos/icon-browser/icon.list

**Comment by dmikushin** [comment]
@ReenigneArcher , yes you are right, one way is to install icon into `/usr/share/icons`. Another way is to embed icon into a binary, and hope that a desktop support its extraction - I don't think it's a popular choice at least due to performance reasons. Please propose a patch how do you envision icon customization in CMake.

**Comment by VENTDEV** [comment]
I'm late to the party, but it would be nice to have the old solution still available. I am not particularly interested in supporting Gnome/KDE and I would rather not have to use C++20/QT6 for something so trivial.

**Comment by dmikushin** [comment]
Hi @VENTDEV  , all you have to do is to clone the repository and check out the commit before Qt6 solution was added. Perhaps, we should add a tag for it. Do let us know how it works for you.

Please note that KDE is being developed according to their own evangelism. From their point of view, I can see that Qt6 is highly more preferable than appindicator. Simply because **KDE is a Qt-based desktop environment**.
